marconi-sidechain/README.md

```diff
@@ -1,37 +1,67 @@
 # marconi-sidechain
 
-`marconi-sidechain` is a lightweight chain follower application for the Sidechain project to index and query specific information from the Cardano blockchain.
+`marconi-sidechain` is a lightweight chain-follower application for the Sidechain project to index and query specific information from the Cardano blockchain.
 The interface for querying the indexed information uses [JSON-RPC](https://www.jsonrpc.org/specification) over HTTP.
 
-See the [architecture documentation](./doc/ARCHITECTURE.adoc) for more information on how this application was build.
+See the [architecture documentation](./doc/ARCHITECTURE.adoc) for more information on how this application was built.
 
 ## Prerequisites
 
-If using `Nix`:
+If using **Nix**:
 
 * [Nix](https://nixos.org/download.html) (`>=2.5.1`)
-  * Enable IOHK's binary cache or else you will build the world! Refer to [this section](../CONTRIBUTING.adoc#how-to-get-a-shell-environment-with-tools) on how to achieve this.
+* Ensure you have the following configuration (eg in `~/.config/nix/nix.conf`):
 
-If *not* using `Nix`:
+```
+experimental-features = nix-command flakes
+accept-flake-config = true
+```
+
+If using **Docker**:
+
+```
+docker pull nixos/nix:2.18.1
+```
+
+If *not* using **Nix** or **Docker**:
 
 * [GHC](https://www.haskell.org/downloads/) (`==8.10.7`)
 * [Cabal](https://www.haskell.org/cabal/download.html) (`>=3.4.0.0`)
 * [cardano-node](https://github.com/input-output-hk/cardano-node/releases/tag/1.35.4) (`==1.35.4`) running on preview testnet, pre-production testnet or mainnet
 
 ## How to build from source
 
-### Cabal build
+### Nix build
 
-TBD
+The `marconi-sidechain` executable is available as a nix flake.
+
+If inside the `marconi` repository, you can run from the top-level:
+
+```
+$ nix build .#marconi-sidechain
+```
```

Comment:
On Mac apple silicon, we need to force a x86 build using `--system x86_64-darwin`. I didn't find a nice way to put it but if could be a nice addition.

Reply:
I see! So we can't build Marconi for `aarch64-darwin`? I didn't realize that.

I'll add an additional paragraph saying that "for now" we can't build native binaries for Apple Silicon and so x86 has to be forced using that arg.

Reply:
Yes, the native silicon doesn't build (I don't remember for which reason). I know the devx team is supposed to work on it.

Reply:
I assume that means the Nix+Cabal option doesn't work either, so I've added notes to both Nix and Nix+Cabal. LMK if Nix+Cabal is OK and I'll take out that note.